pyiceberg/manifest.py

```diff
@@ -1087,18 +1087,15 @@ def add_entry(self, entry: ManifestEntry) -> ManifestWriter:
         return self
 
     def add(self, entry: ManifestEntry) -> ManifestWriter:
-        if entry.sequence_number is not None and entry.sequence_number >= 0:
-            self.add_entry(
-                ManifestEntry.from_args(
-                    snapshot_id=self._snapshot_id, sequence_number=entry.sequence_number, data_file=entry.data_file
-                )
-            )
-        else:
-            self.add_entry(
-                ManifestEntry.from_args(
-                    status=ManifestEntryStatus.ADDED, snapshot_id=self._snapshot_id, data_file=entry.data_file
-                )
+        self.add_entry(
+            ManifestEntry.from_args(
+                status=ManifestEntryStatus.ADDED,
+                snapshot_id=self._snapshot_id,
+                sequence_number=entry.sequence_number if entry.file_sequence_number != UNASSIGNED_SEQ else None,
```

Comment:
```suggestion
                sequence_number=entry.sequence_number if entry.sequence_number != UNASSIGNED_SEQ else None,
```

should we check `sequence_number` instead? 
https://github.com/apache/iceberg-python/blob/38c569b51b909c0b14f07b972a4a4783dbf7868c/pyiceberg/manifest.py#L1049